modules/core/src/core/lib/layer.js

```diff
@@ -362,6 +362,28 @@ export default class Layer {
     }
   }
 
+  // this method overwrittes the current buffer and is used for multi-picking
+  updateInstancePickingColors(excludeColors) {
+    const excludeIds = excludeColors.map(this.decodePickingColor);
+
+    const {instancePickingColors} = this.getAttributeManager().attributes;
+    const {state: attribute, allocedInstances: numInstances} = instancePickingColors;
+
+    const {value, size} = attribute;
+    // add 1 to index to seperate from no selection
+    for (let i = 0; i < numInstances; i++) {
+      const pickingColor = excludeIds.includes(i)
+        ? this.nullPickingColor()
+        : this.encodePickingColor(i);
+      value[i * size + 0] = pickingColor[0];
+      value[i * size + 1] = pickingColor[1];
+      value[i * size + 2] = pickingColor[2];
+    }
+
+    // TODO: Optimize this to use sub-buffer update!
+    this.getSingleModel().setAttributes({instancePickingColors: attribute});
```

Comment:
`getSingleModel()` may return null

Reply:
When? Why?

Reply:
We support multi model layers, so it can happen. As discussed, for now add if statement to make sure we don't crash in this case.